bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -0,0 +1,1242 @@
+
+impl From<&Ed25519Address> for Ed25519AddressDto {
+    fn from(value: &Ed25519Address) -> Self {
+        Self {
+            kind: Ed25519Address::KIND,
+            address: value.to_string(),
+        }
+    }
+}
+
+impl TryFrom<&Ed25519AddressDto> for Ed25519Address {
+    type Error = Error;
+
+    fn try_from(value: &Ed25519AddressDto) -> Result<Self, Self::Error> {
+        value
+            .address
+            .parse::<Ed25519Address>()
+            .map_err(|_| Error::InvalidSyntaxField("address"))
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AssetBalanceDto {
+    id: String, //[u8; Self::LENGTH],
+    balance: u64,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DataPayloadDto {
+    #[serde(rename = "type")]
+    pub kind: u32,
+    data: String,
+}
+
+impl From<&DataPayload> for DataPayloadDto {
+    fn from(value: &DataPayload) -> Self {
+        DataPayloadDto {
+            kind: DataPayload::KIND,
+            data: hex::encode(value.data())
+        }
+    }
+}
+
+impl TryFrom<&DataPayloadDto> for DataPayload {
+    type Error = Error;
+
+    fn try_from(value: &DataPayloadDto) -> Result<Self, Self::Error> {
+        if value.data.len() % 2 != 0 {
+            return Err(Error::InvalidSyntaxField("data"));
+        }
+        match (0..value.data.len())
+            .step_by(2)
+            .map(|i| value.data.get(i..i + 2)
+            .and_then(|sub| u8::from_str_radix(sub, 16).ok()))
+            .collect() {
+                Some(data) => Ok(DataPayload::new(data).map_err(|_| Error::InvalidSyntaxField("data"))?),
+                None => Err(Error::InvalidSyntaxField("data")),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct FpcPayloadDto {
+    #[serde(rename = "type")]
+    pub kind: u32,
+    conflicts: Vec<ConflictDto>,
+    timestamps: Vec<TimestampDto>,
+}
+
+impl From<&FpcPayload> for FpcPayloadDto {
+    fn from(value: &FpcPayload) -> Self {
+        FpcPayloadDto {
+            kind: FpcPayload::KIND,
+            conflicts: value.conflicts().map(|c| c.into()).collect(),
+            timestamps: value.timestamps().map(|t| t.into()).collect(),
+        }
+    }
+}
+
+impl TryFrom<&FpcPayloadDto> for FpcPayload {
+    type Error = Error;
+
+    fn try_from(value: &FpcPayloadDto) -> Result<Self, Self::Error> {
+        let builder = FpcPayload::builder()
+        .with_conflicts(
+            value.conflicts.iter().map(|c| c.try_into())
+            .collect::<Result<Vec<Conflict>, Error>>()?,
+        )
+        .with_timestamps(
+            value.timestamps.iter().map(|c| c.try_into())
+            .collect::<Result<Vec<Timestamp>, Error>>()?,
+        );
+
+        Ok(builder.finish()?)
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ConflictDto {
+    /// Identifier of the conflicting transaction.
+    transaction_id: String,
+    /// The node's opinion value in a given round.
+    opinion: u8,
```

Comment:
Why `opinion` does not have `Opinion` type?

Reply:
Yh, could be, but it's still the same primitive type.